### PR TITLE
Encryption key rotation

### PR DESCRIPTION
**Pull request by graetzer**
### Scope & Purpose

Allow to rotate encryption keys (by re-encrypting the actual internal key)

- [x] Strictly *new functionality* (i.e. a new feature / new option, no need for porting)
- [x] The behavior in this PR can be (and was) *manually tested* (support / qa / customers can test it)
- [x] The behavior change can be verified via automatic tests

### Testing & Verification

This PR adds tests that were used to verify all changes:

- [x] Added new **integration tests** in the recovery testsuite

http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr/8513/

### Documentation

- [x] Added a new section in the *Manual*  https://github.com/arangodb/docs/pull/332

### CODE REVIEW

**Comment by graetzer** [comment]
http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr/8513/

**Comment by graetzer** [comment]
http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr/8520/

**Comment by jsteemann** [comment]
Tests blue

**Comment by graetzer** [comment]
http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr/8978/

File: lib/Basics/FileUtils.cpp [hunk 1]
```diff
@@ -365,19 +365,19 @@ bool copyRecursive(std::string const& source, std::string const& target,
...
+//
+//  // "auto lambda" will not work here
+//  std::function<TRI_copy_recursive_e(std::string const&)> lambda =
+//    [&filter] (std::string const& pathname) -> TRI_copy_recursive_e {
+//    return filter(pathname) ? TRI_COPY_IGNORE : TRI_COPY_COPY;
+//  };
+//
+//  return copyDirectoryRecursive(source, target, lambda, error);
+//
+//} // copyDirectoryRecursive (bool filter())
```
Comment (jsteemann) [inline lib/Basics/FileUtils.cpp]: Let's remove this altogether if not needed!

File: lib/Basics/FileUtils.h [hunk 2]
```diff
@@ -87,9 +87,9 @@ bool copyRecursive(std::string const& source, std::string const& target,
 
 /// @brief will not copy files/directories for which the filter function
 /// returns true (now wrapper for version below with TRI_copy_recursive_e filter)
-bool copyDirectoryRecursive(std::string const& source, std::string const& target,
-                            std::function<bool(std::string const&)> const& filter,
-                            std::string& error);
+//bool copyDirectoryRecursive(std::string const& source, std::string const& target,
+//                            std::function<bool(std::string const&)> const& filter,
+//                            std::string& error);
```
Comment (jsteemann) [inline lib/Basics/FileUtils.h]: same

**Comment by jsteemann** [comment]
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/9018/

**Comment by jsteemann** [comment]
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/9027/

**Pull request merged**